### PR TITLE
docs: add GitHub Actions and GitLab example to CI/CD guide

### PR DESCRIPTION
**Pull request by lizardruss**
### Changes
- Add CI/CD examples for GitHub Actions
- Add GitLab CI/CD example Pipeline

### CODE REVIEW

**Comment by FabianKramm** [comment]
@LukasGentele please take a look

File: docs/pages/guides/ci-cd-pipelines.mdx [hunk 1]
```diff
@@ -162,4 +164,370 @@ RUN chmod +x /bin/loft
...
+
+## Using GitHub Actions
+
+Loft provides the following GitHub Actions for use in workflows:
+- [Setup DevSpace](https://github.com/loft-sh/setup-devspace): Installs the `devspace` CLI
+- [Setup Loft](https://github.com/loft-sh/setup-loft): Installs the `loft` CLI and logs in to the Loft instance
+- [Create Space](https://github.com/loft-sh/create-space): Creates a Loft Space
+- [Delete Space](https://github.com/loft-sh/delete-space): Deletes a Loft Space
+- [Create Virtual Cluster](https://github.com/loft-sh/create-vcluster): Creates a Loft Virtual Cluster
+- [Delete Virtual Cluster](https://github.com/loft-sh/delete-vcluster): Deletes a Loft Virtual Cluster
```
Comment (LukasGentele) [inline docs/pages/guides/ci-cd-pipelines.mdx:177]: We can remove `Loft` here. Space and Virtual Cluster are fine.

Comment (LukasGentele) [reply docs/pages/guides/ci-cd-pipelines.mdx:177]: same on the rest of the page tbh.

File: docs/pages/guides/ci-cd-pipelines.mdx [hunk 2]
```diff
@@ -162,4 +164,370 @@ RUN chmod +x /bin/loft
...
+jobs:
+  e2e:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Loft CLI
+        uses: loft-sh/setup-loft@main
+        with:
+          url: ${{ secrets.LOFT_URL }}
+          access-key: ${{ secrets.LOFT_ACCESS_KEY }}
+      - name: Create PR Space
```
Comment (LukasGentele) [inline docs/pages/guides/ci-cd-pipelines.mdx:211]: Create Space for PR

File: docs/pages/guides/ci-cd-pipelines.mdx [hunk 3]
```diff
@@ -162,4 +164,370 @@ RUN chmod +x /bin/loft
...
+
+Loft provides the following GitHub Actions for use in workflows:
+- [Setup DevSpace](https://github.com/loft-sh/setup-devspace): Installs the `devspace` CLI
+- [Setup Loft](https://github.com/loft-sh/setup-loft): Installs the `loft` CLI and logs in to the Loft instance
+- [Create Space](https://github.com/loft-sh/create-space): Creates a Loft Space
+- [Delete Space](https://github.com/loft-sh/delete-space): Deletes a Loft Space
+- [Create Virtual Cluster](https://github.com/loft-sh/create-vcluster): Creates a Loft Virtual Cluster
+- [Delete Virtual Cluster](https://github.com/loft-sh/delete-vcluster): Deletes a Loft Virtual Cluster
+
+### Using a Space for Pull Requests
```
Comment (LukasGentele) [inline docs/pages/guides/ci-cd-pipelines.mdx:179]: Spaces for Pull Requests

File: docs/pages/guides/ci-cd-pipelines.mdx [hunk 4]
```diff
@@ -162,4 +164,370 @@ RUN chmod +x /bin/loft
...
+        run: kubectl rollout status -n pr-${{ github.event.pull_request.number }}-${{ github.sha }}-${{ github.run_id }} deployments/my-app
+      - name: Run Tests
+        run: make e2e
+      - name: Delete PR Space
+        uses: loft-sh/delete-space@main
+        with:
+          name: pr-${{ github.event.pull_request.number }}-${{ github.sha }}-${{ github.run_id }}
+```
+
+#### Step by Step Explanation
```
Comment (LukasGentele) [inline docs/pages/guides/ci-cd-pipelines.mdx:227]: No level-4 headlines here (also rest of the page)

Comment (LukasGentele) [reply docs/pages/guides/ci-cd-pipelines.mdx:227]: Convert to regular text (see devspace docs: https://devspace.sh/cli/docs/configuration/images/image-tagging#image-image-repository)

File: docs/pages/guides/ci-cd-pipelines.mdx [hunk 5]
```diff
@@ -162,4 +164,370 @@ RUN chmod +x /bin/loft
...
+1. The [Setup DevSpace](https://github.com/loft-sh/setup-devspace) action installs the DevSpace CLI.
+2. The [Setup Loft](https://github.com/loft-sh/setup-loft) action is used to install the Loft CLI and login using the provided `url` and `access-key`.
+3. The [Create Space](https://github.com/loft-sh/create-space) action is used to create a unique space using information about the pull request. This will automatically configure the kube context for the following steps. Additionally, we have enabled the `auto-cleanup` option, which will delete the space after the job completes.
+4. Finally we use `devspace run e2e` to perform the needed steps to deploy and test `my-app`.
+
+</TabItem>
+</Tabs>
+
+### Using a Virtual Cluster for Pull Requests
```
Comment (LukasGentele) [inline docs/pages/guides/ci-cd-pipelines.mdx:317]: Virtual Clusters for Pull Requests

File: docs/pages/guides/ci-cd-pipelines.mdx [hunk 6]
```diff
@@ -162,4 +164,370 @@ RUN chmod +x /bin/loft
...
+2. The [Setup Loft](https://github.com/loft-sh/setup-loft) action is used to install the Loft CLI and login using the provided `url` and `access-key`.
+3. The [Create Virtual Cluster](https://github.com/loft-sh/create-vcluster) action is used to create a unique virtual cluster using information about the pull request. This will automatically configure the kube context for the following steps. Additionally, we have enabled the `auto-cleanup` option, which will delete the virtual cluster after the job completes.
+4. Finally we use `devspace run e2e` to perform the needed steps to deploy and test `my-app`.
+
+</TabItem>
+
+
+</Tabs>
+
+## Using GitLab CI/CD Pipelines
```
Comment (LukasGentele) [inline docs/pages/guides/ci-cd-pipelines.mdx:459]: \### GitLab CI/CD

File: docs/pages/guides/ci-cd-pipelines.mdx [hunk 7]
```diff
@@ -162,4 +164,370 @@ RUN chmod +x /bin/loft
 
 ### Login with Access Keys
 
-<LoginAccessKeysFragment/>
\ No newline at end of file
+<LoginAccessKeysFragment/>
+
+## Using GitHub Actions
```
Comment (LukasGentele) [inline docs/pages/guides/ci-cd-pipelines.mdx:169]: \## Integrations
\### GitHub Actions

=> Turn current h3 headlines into h4 headlines on the entire page.

File: docs/pages/guides/ci-cd-pipelines.mdx [hunk 8]
```diff
@@ -10,8 +10,10 @@ import LoginAccessKeysFragment from '../fragments/profile/login-access-key.mdx'
 
 When using Loft in a CI/CD pipeline to create namespaces and virtual cluster, there are couple of things to consider:
 1. You might want to generate your kube configs manually
-2. You may want to use the [offical `loftsh/loft-ci` image](#container-image) in a containerized CI/CD pipeline
+2. You may want to use the [offical `loftsh/loft-ci` image](#using-the-container-image) in a containerized CI/CD pipeline
 3. You definitely want to [authenticate using Access Keys](#authentication) in either case
+4. If using GitHub Actions for CI/CD see the section on [Using GitHub Actions](#using-github-actions)
+5. If using GitLab for CI/CD  see [Using GitLab CI/CD Pipelines](#using-gitlab-cicd-pipelines)
 
 ## Accessing Loft
```
Comment (LukasGentele) [inline docs/pages/guides/ci-cd-pipelines.mdx:18]: Let's rename this to: Authentcation & Cluster Access

File: docs/pages/guides/ci-cd-pipelines.mdx [hunk 9]
```diff
@@ -162,4 +164,370 @@ RUN chmod +x /bin/loft
...
+</TabItem>
+
+
+</Tabs>
+
+## Using GitLab CI/CD Pipelines
+
+When using Loft with GitLab you can use the [official `loft-ci` image](https://hub.docker.com/r/loftsh/loft-ci) as either a base image or directly. See [Using the Container Image](#using-the-container-image) for details on the available tools. If additional tooling is needed for your CI/CD process, a custom image can be created.
+
+### Using Spaces for Merge Requests
```
Comment (LukasGentele) [inline docs/pages/guides/ci-cd-pipelines.mdx:463]: -Using

File: docs/pages/guides/ci-cd-pipelines.mdx [hunk 10]
```diff
@@ -162,4 +164,370 @@ RUN chmod +x /bin/loft
...
+
+#### Step by Step Explanation
+1. The `loftsh/loft-ci` image is used for all pipeline jobs and provides the `loft` CLI, the `devspace` CLI, and `kubectl`.
+2. The `before_script` first logs in to loft using the `$LOFT_URL` and `$LOFT_ACCESS_KEY` variables that you defined in GitLab. [See the GitLab docs for more information](https://docs.gitlab.com/ee/ci/variables/)
+3. The `before_script` then creates a space using [predefined GitLab variables](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html) to create a unique name for the space.
+4. Next `before_script` installs some additional tooling needed to run the end-to-end tests. For more complex scenarios creating a custom image is recommended.
+5. Then the `script` section uses `kubectl` to deploy the application to the space and waits for the `my-app` deployment to become ready. `make` is then used to run an end-to-end test suite.
+6. Finally `after_script` deletes the space. By using `after_script` we can ensure the space is deleted even if the tests fail.
+
+### Using Virtual Clusters for Merge Requests
```
Comment (LukasGentele) [inline docs/pages/guides/ci-cd-pipelines.mdx:499]: -Using

**Review by LukasGentele — CHANGES_REQUESTED** [review]

**Review by LukasGentele — APPROVED** [review]

**Pull request merged**